cpp/tests/prims/mg_per_v_transform_reduce_incoming_outgoing_e.cu

```diff
@@ -174,67 +191,124 @@ class Tests_MGPerVTransformReduceIncomingOutgoingE
+                                            mg_src_prop.view(),
+                                            mg_dst_prop.view(),
+                                            cugraph::edge_dummy_property_t{}.view(),
+                                            e_op_t<vertex_t, result_t>{},
+                                            property_initial_value,
+                                            cugraph::reduce_op::minimum<result_t>{},
+                                            cugraph::get_dataframe_buffer_begin(in_results[i]));
+          break;
+        case reduction_type_t::MAXIMUM:
+          per_v_transform_reduce_incoming_e(*handle_,
+                                            mg_graph_view,
+                                            mg_src_prop.view(),
+                                            mg_dst_prop.view(),
+                                            cugraph::edge_dummy_property_t{}.view(),
+                                            e_op_t<vertex_t, result_t>{},
+                                            property_initial_value,
+                                            cugraph::reduce_op::maximum<result_t>{},
+                                            cugraph::get_dataframe_buffer_begin(in_results[i]));
+          break;
+        default: FAIL() << "should not be reached.";
+      }
 
-    per_v_transform_reduce_incoming_e(
-      *handle_,
-      mg_graph_view,
-      mg_src_prop.view(),
-      mg_dst_prop.view(),
-      cugraph::edge_dummy_property_t{}.view(),
-      [] __device__(auto src, auto dst, auto src_property, auto dst_property, thrust::nullopt_t) {
-        if (src_property < dst_property) {
-          return src_property;
-        } else {
-          return dst_property;
-        }
-      },
-      property_initial_value,
-      cugraph::get_dataframe_buffer_begin(in_result));
+      if (cugraph::test::g_perf) {
+        RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+        handle_->get_comms().barrier();
+        hr_timer.stop();
+        hr_timer.display_and_clear(std::cout);
+      }
 
-    if (cugraph::test::g_perf) {
-      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
-      handle_->get_comms().barrier();
-      hr_timer.stop();
-      hr_timer.display_and_clear(std::cout);
-    }
+      out_results.push_back(cugraph::allocate_dataframe_buffer<result_t>(
+        mg_graph_view.local_vertex_partition_range_size(), handle_->get_stream()));
 
-    if (cugraph::test::g_perf) {
-      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
-      handle_->get_comms().barrier();
-      hr_timer.start("MG per_v_transform_reduce_outgoing_e");
-    }
+      if (cugraph::test::g_perf) {
+        RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+        handle_->get_comms().barrier();
+        handle_->get_comms().barrier();
```

Comment:
Isn't second barrier redundant?

Reply:
Yes and fixed, I guess this happened while resolving merge conflicts.